src/stream_base.cc

```diff
@@ -100,94 +100,118 @@ int StreamBase::Writev(const FunctionCallbackInfo<Value>& args) {
 
   Local<Object> req_wrap_obj = args[0].As<Object>();
   Local<Array> chunks = args[1].As<Array>();
+  bool all_buffers = args[2]->BooleanValue();
 
-  size_t count = chunks->Length() >> 1;
+  size_t count;
+  if (all_buffers)
+    count = chunks->Length();
+  else
+    count = chunks->Length() >> 1;
 
   MaybeStackBuffer<uv_buf_t, 16> bufs(count);
+  uv_buf_t* buf_list = *bufs;
 
-  // Determine storage size first
   size_t storage_size = 0;
-  for (size_t i = 0; i < count; i++) {
-    storage_size = ROUND_UP(storage_size, WriteWrap::kAlignSize);
-
-    Local<Value> chunk = chunks->Get(i * 2);
-
-    if (Buffer::HasInstance(chunk))
-      continue;
-      // Buffer chunk, no additional storage required
-
-    // String chunk
-    Local<String> string = chunk->ToString(env->isolate());
-    enum encoding encoding = ParseEncoding(env->isolate(),
-                                           chunks->Get(i * 2 + 1));
-    size_t chunk_size;
-    if (encoding == UTF8 && string->Length() > 65535)
-      chunk_size = StringBytes::Size(env->isolate(), string, encoding);
-    else
-      chunk_size = StringBytes::StorageSize(env->isolate(), string, encoding);
-
-    storage_size += chunk_size;
-  }
+  uint32_t bytes = 0;
+  size_t offset;
+  AsyncWrap* wrap;
+  WriteWrap* req_wrap;
+  int err;
 
-  if (storage_size > INT_MAX)
-    return UV_ENOBUFS;
+  if (!all_buffers) {
+    // Determine storage size first
+    for (size_t i = 0; i < count; i++) {
+      storage_size = ROUND_UP(storage_size, WriteWrap::kAlignSize);
 
-  AsyncWrap* wrap = GetAsyncWrap();
-  // NOTE: All tests show that GetAsyncWrap() never returns nullptr here. If it
-  // can then replace the CHECK_NE() with if (wrap != nullptr).
-  CHECK_NE(wrap, nullptr);
-  env->set_init_trigger_id(wrap->get_id());
-  WriteWrap* req_wrap = WriteWrap::New(env,
-                                       req_wrap_obj,
-                                       this,
-                                       AfterWrite,
-                                       storage_size);
+      Local<Value> chunk = chunks->Get(i * 2);
 
-  uint32_t bytes = 0;
-  size_t offset = 0;
-  for (size_t i = 0; i < count; i++) {
-    Local<Value> chunk = chunks->Get(i * 2);
+      if (Buffer::HasInstance(chunk))
```

Comment:
Actually in master `HasInstance()` uses `IsArrayBufferView()` while v7.x and v6.x use `IsUint8Array()`. Ideally this could be backported more easily if we kept `HasInstance()`?